lib/reporter-proxy.js

```diff
@@ -0,0 +1,63 @@
+const pjson = require('../package.json');
+
+// this class allows us to call reporter methods
+// before the reporter is actually loaded, since we don't want to
+// assume that the metrics package will load before the GitHub package.
+
+// maybe this should be an object instead of a class.
+// IDK.
+class ReporterProxy {
+  constructor() {
+    this.reporter = null;
+    this.events = [];
+    this.timings = [];
+    this.counters = [];
+    this.gitHubPackageVersion = pjson.version;
+  }
+
+  // function that is called after the reporter is actually loaded, to
+  // set the reporter and send any data that have accumulated while it was loading.
+  setReporter(reporter) {
+    this.reporter = reporter;
+
+    this.events.forEach(customEvent => {
+      this.reporter.addCustomEvent(customEvent.eventType, customEvent.event);
+    });
+
+    this.timings.forEach(timing => {
+      this.reporter.addTiming(timing.eventType, timing.durationInMilliseconds, timing.metadata);
+    });
+
+    this.counters.forEach(counterName => {
+      this.reporter.incrementCounter(counterName);
+    });
+  }
+}
+
+export const reporterProxy = new ReporterProxy();
+
+export const incrementCounter = function(counterName) {
```

Comment:
I wasn't sure if `export function name() { .. }` retained the execution context of the module where the function was defined, which is necessary because I'm using closures. but I suppose I could bind if I need to.  Consistency is good, I'll try to switch to the other style. Thanks!